src/js/oncoprint.js

```diff
@@ -0,0 +1,125 @@
+var Track = require('./track').Track;
+var _ = require('underscore');
+var d3 = require('d3');
+var $ = require('jquery');
+module.exports = {};
+
+// TODO: use self everywhere
+
+var defaultOncoprintConfig = {
+	cell_width: 10,
+	cell_padding: 3,
+	render: 'table',
+};
+
+function Oncoprint(container_selector_string, config) {
+	var self = this;
+	self.table;
+	self.config = $.extend({}, defaultOncoprintConfig, config || {});
+
+	self.id_order = [];
+	self.track_order = []; 
+	self.tracks = {};
+	self.ids = {};
+
+	if (self.config.render === 'table') {
+		self.renderer = new OncoprintTableRenderer(container_selector_string, self);
+	}
+
+	self.setCellWidth = function(w) {
+		self.config.cell_width = w;
+		// trigger event
+		$(self).trigger('set_cell_width.oncoprint');
+	};
+
+	self.setCellPadding = function(p) {
+		self.config.cell_padding = p;
+		// trigger event
+		$(self).trigger('set_cell_padding.oncoprint');
+	};
+
+	self.sortOnTrack = function(trackName, dataCmp) {
+		// sort ids using given comparator, by delegating to the track
+		self.id_order = self.tracks[trackName].getDatumIds(dataCmp);
+		// trigger event
+		$(self).trigger('sort.oncoprint', {id_order:this.id_order});
+	};
+
+	self.moveTrack = function(trackName, newPosition) {
+		// remove from old position in order and place it in new position
+		newPosition = Math.min(self.track_order.length-1, newPosition);
+		newPosition = Math.max(0, newPosition);
+		var oldPosition = self.track_order.indexOf(trackName);
+		self.track_order.splice(oldPosition, 1);
+		self.track_order.splice(newPosition, 0, trackName);
+		// trigger event
+		$(self).trigger('move_track.oncoprint', {track: self.tracks[trackName], new_position: newPosition, track_order: this.track_order});
+	};
+
+	self.appendTrack = function(name, data, config) {
+		// two tracks with same name not allowed
+		if (name in self.tracks) {
```

Comment:
You're right - I should stop now before it gets entrenched.

I'll just change ```self``` to ```var```